### ISSUE TITLE
fontawesome.com pages take 7+ seconds to load

### ISSUE THREAD

**Issue by AndrewBelt** [FortAwesome/Font-Awesome#16230]
**Describe the bug**
Most pages on fontawesome.com, but particularly https://fontawesome.com/icons take 7 seconds to load on my desktop, 15 seconds on my laptop, and 30+ seconds on mobile.

**To Reproduce**
Cache cleared or not, go to https://fontawesome.com/icons in your browser and wait for the icons to display. 

**Expected behavior**
7 seconds seems abnormally high. I'd expect <2s would be reasonable for a website like this.

**Screenshots**
If the Network Inspector on Firefox Linux is open, it looks like this after the page is open.

![2020-02-21-111904_1600x900_scrot](https://user-images.githubusercontent.com/338179/75051869-a9493500-549c-11ea-9b67-af348d27777e.png)

The timing chart shows there there is a lot of time between groups of requests. First the main HTML at 0ms, then JS, PNG, and JSON at ~1s, then Google fonts at ~5s, then Stripe POST requests at ~7s.

It seems that network latency isn't the problem here. It's probably doing a lot of local computation between each of the "groups".

**Version and implementation**
Browser and version: I have tested on Linux, Windows, Mac, and Android with Firefox and Chrome.

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

I know that the icon gallery is not fast, but let's ask @robmadole if we can do something in the short term or a refactor has been planned for the future

**Comment by robmadole** [comment]
Yep, certainly the page is slow but we aren't seeing the same kind of load times that you are reporting. The icons page loads on Desktop Chrome in less than 2 seconds and less than 3 on mobile.

We'll be focusing on performance of the site, most likely, this year.

**Comment by AndrewBelt** [comment]
Awesome, glad to hear! I'm not too bothered by it myself, but if it's the case that other people are experiencing it, performance enhancement would be a great improvement when navigating for icons. An important detail here is that pageload time doesn't appear to be any faster when navigating around the website, even when most assets are cached locally.

Just a random thought, feel free to ignore:
Making a website out of mostly dynamic JS-generated DOM from a JSON API is almost always going to be slower than a server-generated HTML page. But for a web app, this is acceptable because the page has a lifetime over your entire web app experience until you close the tab. However, the Font Awesome website loads new pages when clicking links instead of rebuilding the DOM with JS without reloading the page, etc. In other words, fontawesome.com has the slowness of a JS-generated DOM while *not* having the benefit of a JS-generated DOM, namely zero-latency "fake pageloads".

I say this hypocritically because I designed a web app exactly like this 2 years ago, and I'm working on overhauling it as server-generated web app. Preliminary tests show that load times will fall from 5s to 500ms.
